src/doc/en/developer/workflows.rst

```diff
@@ -2,200 +2,176 @@
+the Sage repo.
 
-The first step is to create your own fork of the Sage repository;
-simply click "Fork" on the `Sage GitHub repository
-<https://github.com/sagemath/sage>`_. Then add it as one of the
-remotes to your local Sage repository. In the following, we will use
-the label "github" for this remote repository, though you are of
-course free to use a different one::
 
-    $ git remote add github git@github.com:github_user_name/sage.git
-    $ git remote -v
-    github      git@github.com:github_user_name/sage.git (fetch)
-    github      git@github.com:github_user_name/sage.git (push)
-    trac        git@trac.sagemath.org:sage.git (fetch)
-    trac        git@trac.sagemath.org:sage.git (push)
-    $ git fetch github
-    remote: Counting objects: 107, done.
-    remote: Compressing objects: 100% (63/63), done.
-    remote: Total 74 (delta 41), reused 40 (delta 10)
-    Unpacking objects: 100% (74/74), done.
-    From github.com:github_user_name/sage
-    * [new branch]      master     -> github/master
+.. _section-workflows-finish:
 
+Finishing It Up
+===============
 
-Develop
--------
+It is common to go through a few iterations of commits before you
+push the branch, and you will probably also have pushed your branch a few
+times before your branch is ready for review.
 
-You now use the github repository to develop your ticket branch; First
-create a new branch::
+Once you are happy with the changes you pushed, they must be
+reviewed by someone else before they can be included in the next
+release of Sage. To mark your PR as ready for review, you should
+set it to ``needs review`` status.
 
-    $ git checkout -b my_branch --track github/master
-    Branch my_branch set up to track remote branch master from github.
-    Switched to a new branch 'my_branch'
-    $ git push github my_branch
-    Total 0 (delta 0), reused 0 (delta 0)
-    To git@github.com:github_user_name/sage.git
-     * [new branch]      my_branch -> my_branch
 
-Because of the ``--track`` option, the ``git pull`` command will
-default to downloading your coauthor's changes from your github
-branch. Alternatively, you can create a new branch on your fork's
-GitHub webpage.
+.. _section-workflows-merge:
 
-At this point you can use the GitHub workflow that you prefer. In
-particular, your choices are
+Merging Upstream develop Branch
+===============================
 
-* Give your coauthors write permissions to your github fork. Every
-  author edits/commits to their own local copy and they jointly push
-  to your github branch.
+It commonly happens that ``develop`` branch at the remote ``upstream`` was updated and you need to merge the upstream changes to your local branch. Then you do::
 
-* Have every coauthor create their own fork and send you (the lead
-  author) pull requests to your GitHub fork.
+    [alice@localhost sage]$ git checkout develop
+    ....
+    [alice@localhost sage]$ git pull upstream
+    ....
```

Comment:
Not just `git fetch upstream develop`?

Reply:
This assumes that you are on `local_working_branch`. You now want to pull the upstream ``develop`` to your local ``develop`` branch.

`git fetch upstream develop` doesn't do it. Do you mean something else?

Reply:
Do you mean `git fetch upstream develop:develop `?

Reply:
I replaced with `git fetch upstream develop:develop`.

Reply:
Forgot to say that then `git merge develop` needs to be changed to `git merge upstream/develop`.

But it's OK as is.